packages/desktop-client/src/components/reports/CustomSidebar.js

```diff
@@ -0,0 +1,417 @@
+import React from 'react';
+
+import * as monthUtils from 'loot-core/src/shared/months';
+
+import { theme } from '../../style';
+import Button from '../common/Button';
+import Select from '../common/Select';
+import Text from '../common/Text';
+import View from '../common/View';
+import { Checkbox } from '../forms';
+
+import CategorySelector from './CategorySelector';
+import {
+  validateStart,
+  validateEnd,
+  getLatestRange,
+  getFullRange,
+} from './Header';
+
+export function CustomSidebar({
+  start,
+  setStart,
+  end,
+  setEnd,
+  dateRange,
+  setDateRange,
+  dateRangeOptions,
+  dateRangeLine,
+  allMonths,
+  graphType,
+  setGraphType,
+  setViewSplit,
+  typeDisabled,
+  setTypeDisabled,
+  split,
+  setSplit,
+  splitOptions,
+  type,
+  setType,
+  typeOptions,
+  mode,
+  setMode,
+  empty,
+  setEmpty,
+  hidden,
+  setHidden,
+  uncat,
+  setUncat,
+  categories,
+  selectedCategories,
+  setSelectedCategories,
+}) {
+  function onChangeDates(start, end) {
+    setStart(start);
+    setEnd(end);
+  }
+
+  function onChangeMode(cond) {
+    setMode(cond);
+    if (cond === 'time') {
+      if (graphType === 'TableGraph') {
+        setTypeDisabled([0]);
+      } else {
+        setTypeDisabled([3]);
+        if ([3].includes(type)) {
+          setType(1);
+        }
+      }
+      if (graphType === 'BarGraph') {
+        setGraphType('StackedBarGraph');
+      }
+      if (['AreaGraph', 'DonutGraph'].includes(graphType)) {
+        setGraphType('TableGraph');
+        setViewSplit(false);
+      }
+      if ([5, 6].includes(split)) {
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏these numbers scattered around the codebase are very cryptic. I have no idea what they are meant to represent just by looking at them.

Could we instead use a constant variable or enum?

Reply:
Done.